### PR TITLE
migrate to 2.0 patch 1

### PR DESCRIPTION
**Pull request by PlexPt**
https://github.com/tburrows13/factorio-2.0-mod-porting-guide#prototype-stage

### CODE REVIEW

**Comment by PlexPt** [comment]
now it can be loaded

File: prototypes/entities/apple-farm.lua [hunk 1]
```diff
@@ -14,40 +14,43 @@ data:extend{
         fluid_boxes = {
             {
                 production_type = "input",
-                base_area = 10,
-                height = 2,
-                base_level = -1,
+                volume = 2000, --  ：base_area * height * 1000 (10 * 2 * 100)
                 pipe_connections = {
-                    { type = "input", position = { 7.3, 0.0 } },
+                    { direction = defines.direction.east, flow_direction = "input", position = { 6.8, 0.0 } }, -- 根据位置调整方向
```
Comment (bahrmichael) [inline prototypes/entities/apple-farm.lua:19]: I won't require a change here, but please use English for future comments.

File: prototypes/technologies/tech.lua [hunk 2]
```diff
@@ -217,7 +217,7 @@ data:extend{
         mod = "Tycoon",
         icon = "__tycoon__/graphics/technology/residential-housing.png",
         icon_size = 256,
-        prerequisites = { "steel-processing", "optics", "tycoon-meat-processing", "tycoon-baking", "tycoon-milking", "tycoon-bottling" },
+        prerequisites = { "steel-processing", "tycoon-meat-processing", "tycoon-baking", "tycoon-milking", "tycoon-bottling" },
```
Comment (bahrmichael) [inline prototypes/technologies/tech.lua:220]: Was optics renamed/dropped in vanilla Factorio, or is there another reason for removing it here?

**Comment by PlexPt** [comment]
dropped



---Original---
From: "Michael ***@***.***&gt;
Date: Tue, Dec 17, 2024 06:04 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [bahrmichael/factorio-tycoon] migrate to 2.0 patch 1 (PR #364)




 
@bahrmichael commented on this pull request.
 
 
In prototypes/technologies/tech.lua:
 &gt; @@ -217,7 +217,7 @@ data:extend{          mod = "Tycoon",          icon = "__tycoon__/graphics/technology/residential-housing.png",          icon_size = 256, -        prerequisites = { "steel-processing", "optics", "tycoon-meat-processing", "tycoon-baking", "tycoon-milking", "tycoon-bottling" }, +        prerequisites = { "steel-processing", "tycoon-meat-processing", "tycoon-baking", "tycoon-milking", "tycoon-bottling" },  
Was optics renamed/dropped in vanilla Factorio, or is there another reason for removing it here?

File: prototypes/technologies/tech.lua (hunk 2 above)
Comment (PlexPt) [reply prototypes/technologies/tech.lua:220]: dropped

Comment (PlexPt) [reply prototypes/technologies/tech.lua:220]: it is be merged to lamp

**Review by bahrmichael — APPROVED** [review]

**Pull request merged**